### DISCUSSION TITLE
OFFICE_ACTION npl_class: Parsing and Consolidation ?

### DISCUSSION THREAD

**Discussion by cverluise** [cverluise/PatCit#24] [Projects & Ideas]
Hello,

Since `v0.2-npl`, we have introduced a OFFICE_ACTION `npl_class`

### Question

How should this class of citations be structured and consolidated?

### Food for thoughts
Below is a sample of 100 citations for this class.

<details>

```sql
WITH
  tmp AS (
  SELECT
    npl_publn_id
  FROM
    `npl-parsing.patcit.v02_npl`
  WHERE
    npl_class="OFFICE_ACTION")
SELECT
  npl_biblio
FROM
  `usptobias.patstat.tls214` AS tls214,
  tmp
WHERE
  tmp.npl_publn_id = tls214.npl_publn_id
LIMIT 100 
```

Row | npl_biblio |  
-- | -- | --
1 | Office Action Response for KR Application No. 10-2013-7015884, filed Aug. 27, 2014, 28 pages. |  
2 | 'Non-Final Office Action', U.S. Appl. No. 10/396,653, (Sep. 19, 2006),24 pages. |  
3 | Official Action for European Patent Application No. 07021150, dated May 6, 2010. |  
4 | U.S. Appl. No. 14/639,310, Aug. 13, 2015, Office Action. |  
5 | Office Action for U.S. Appl. No. 13/087,239, filed Apr. 14, 2011, dated May 22, 2013. |  
6 | Final Office Action dated Jul. 22, 2008; U.S. Appl. No. 10/282,289, filed Oct. 28, 2002; 10 pages. |  
7 | First Office Action issued by the State Intellectual Property Office of People's Republic of China in corresponding CN Application No. 201080037669.8, dated Nov. 18, 2013. |  
8 | U.S. Appl. No. 13/177,341, Response filed Aug. 20, 2012 to Non Final Office Action mailed May 18, 2012, 11 pgs. |  
9 | In the U.S. Patent and Trademark Office, Notice of Allowance in re: U.S. Appl. No. 12/806,356, dated Jul. 11, 2011, 1 page. |  
10 | Notice of Reasons for Rejection from Japanese Patent Application No. 2011-533175, dated Apr. 2, 2013. |  
11 | ' U.S. Appl. No. 11/462,281, Response filed Dec. 29, 2009 to Non Final Office Action mailed Oct. 1, 2009', 11 pgs. |  
12 | Jun. 4, 2008 communication from our foreign associate forwarding Official Communication. |  
13 | IP Australia Office Action dated Jul. 2, 2010 for Australian Patent Application No. 2005-319965, corresponding to U.S. Appl. No. 12/098,263 issued on Jul. 1, 2010. |  
14 | United States Patent and Trademark Office, U.S. Appl. No. 12/685,843, Office Action, Aug. 19, 2010. |  
15 | Non-Final Office Action from U.S. Appl. No. 12/773,781, dated Sep. 23, 2010. |  
16 | Office Action dated Nov. 8, 2011, U.S. Appl. No. 12/745,081. |  
17 | Preliminary Amendment dated Jun. 23, 1997, U.S. Appl. No. 08/881,176, 4 pages. |  
18 | U.S. Appl. No. 10/374,000, Response to Office Action, filed May 12, 2006, 7 pages. |  
19 | 'Non-Final Office Action', U.S. Appl. No. 12/244,545, (Mar. 27, 2012),18 pages. |  
20 | Non-Final Office Action mailed on Feb. 14, 2014, for U.S. Appl. No. 13/498,099, filed Jan. 14, 2013, 20 pages. |  
21 | Office Action issued in Eurasian Patent Application No. 201270371 on Sep. 10, 2015; 5 pages. |  
22 | Office Action issued Sep. 4, 2014 in Chinese Patent Application No. 201110254405.4 (with English language translation). |  
23 | Notice of Allowance and Fee(s) Due for U.S. Appl. No. 13/234,421, dated Sep. 21, 2012. |  
24 | Office Action from U.S. Appl. No. 10/851,471 dated Mar. 27, 2006. |  
25 | 'U.S. Appl. No. 12/987,584, Response filed Sep. 6, 2012 to Non Final Office Action mailed Jun. 6, 2012', 12 pgs. |  
26 | Japanese office action and English translation for JP application No. 2005-539859, dated Jul. 27, 2007 (13 pages). |  
27 | Office Action dated Mar. 7, 2007 cited in U.S. Appl. No. 11/254,264. |  
28 | Non-final Office action regarding U.S. Appl. No. 11/617,497, dated Jun. 26, 2009. |  
29 | U.S. Appl. No. 11/198,606, mailed Feb. 24, 2009, Office Action. |  
30 | U.S. Office Action re U.S. Appl. No. 11/585,230, dated Apr. 5, 2010. |  
31 | Non-Final Office Action mailed on Oct. 12, 2010 in U.S. Appl. No. 12/397,855. |  
32 | Non-Final Office Action from U.S. Appl. No. 09/410,132, mailed on Nov. 21, 2000. |  
33 | Respondent Carsem, Inc.'s Preliminary Invalidity Charts U.S. Patent No. 6,630,728 (Jun. 24, 1998) v. ESEC U.S. Patent No. 5,417,905 (Aug. 4, 1992). |  
34 | Email regarding Office Action 1 (OA1) for ID Application No. W00200700972, dated Jan. 11, 2010, 2 pp. |  
35 | Japanese Office Action, dated Dec. 6, 2011, issued in corresponding Japanese Patent Application No. 2006-305987, with English translation. Total 3 pages. |  
36 | Nonfinal Office Action dated May 11, 2011, in U.S. Appl. No. 12/560,297, filed Sep. 15, 2009, entitled 'Network Intrusion Detection Visualization'. |  
37 | Brayton, John Joseph, 'Office Action re U.S. Appl. No. 13/193,345', Jun. 6, 2013, p. 8, Published in: US. |  
38 | European Patent Office, Communication of a Notice of Opposition issued in Application No. 09742120.0 dated Jun. 26, 2014. |  
39 | U.S. Appl. No. 12/563,798, Office Action mailed Mar. 4, 2011. |  
40 | Final Office Action for U.S. Appl. No. 11/527,246, mailed Mar. 1, 2011. |  
41 | US Ex Parte Quayle Action in U.S. Appl. No. 13/190,143 DTD Aug. 23, 2013. |  
42 | Final Office Action mailed Dec. 13, 2013 in related U.S. Appl. No. 13/113,826. |  
43 | German Office Action for Priority Application DE 102013220663.1 dated May 27, 2014. |  
44 | 'U.S. Appl. No. 13/547,455, Non Final Office Action mailed Oct. 22, 2015', 24 pgs. |  
45 | Final Office Action Mail Date Feb. 13, 2008; U.S. Appl. No. 11/481,194. |  
46 | Office Action mailed Jan. 22, 2015 for U.S. Appl. No. 13/182,914. |  
47 | Non-Final Office Action dated Jul. 3, 2014 for U.S. Appl. No. 13/788,061, filed Mar. 7, 2013 (inventors David Colvin et al.). |  
48 | Notice on the First Office Action issued Oct. 19, 2015 in corresponding Chinese Application No. 201380047592.6 (with English translation). |  
49 | Japanese Office Action (w/translation) issued in application No. 2016-006559, dated Aug. 29, 2016 (3 pgs). |  
50 | Office Action for U.S. Appl. No. 11/964,182, dated Jul. 12, 2011, 12 pages. |  
51 | Office Action dated May 19, 2015, from the Japanese Patent Office in related Japanese Patent Application No. 2014-014941, and English translation thereof. |  
52 | Response to Final Office Action dated Mar. 1, 2013, in connection with related U.S. Appl. No. 12/787,281, filed May 25, 2010. |  
53 | Communication dated Sep. 4, 2014, issued by the State Intellectual Property Office of the People's Republic of China in counterpart Application No. 201280002220.7. |  
54 | Office Action dated Aug. 5, 2014 in related Australian Patent Application serial No. 2014200495, 4 pages. |  
55 | U.S. Final Office Action dated Dec. 13, 2006 issued in U.S. Appl. No. 10/461,451. |  
56 | U.S. Appl. No. 13/341,760 Office Action dated Nov. 20, 2013. |  
57 | 'U.S. Appl. No. 11/517,886, Non-Final Office Action mailed Aug. 6, 2009', 12 pgs. |  
58 | Final Office Action received for U.S. Appl. No. 12/085,389, mailed on Nov. 17, 2009, 14 pages. |  
59 | Restriction Requirement mailed on Jul. 27, 2011 for U.S. Appl. No. 12/418,094, filed Apr. 3, 2009. |  
60 | 'Chinese Application Serial No. 201180055309.5, Voluntary Amendment filed Aug. 23, 2013', w/English Translation, 13 pgs. |  
61 | Excerpted file history of U.S. Appl. No. 13/383,658: USPTO Office action mailed Sep. 27, 2013 for U.S. Appl. No. 13/383,658, The United States Patent and Trademark Office, Alexandria, VA; and Applicants' Amendment and Reply filed Dec. 2, 2013. |  
62 | M. Hachman, “Bluetooth 4.1 Prepares Headsets and More to Connect to the 'Net”, online publication dated Dec. 4, 2013 (http://www.techhive.com/category/holiday/. |  
63 | Office Action for U.S. Appl. No. 13/719,947 dated Apr. 1, 2013, pp. 1-15. |  
64 | USPTO, Office Action for U.S. Appl. No. 12/879,004 mailed Feb. 25, 2013. |  
65 | Office Action for U.S. Appl. No. 10/644,060, mailed on Jun. 3, 2005; 11 pages. |  
66 | 'U.S. Appl. No. 12/770,890, Notice of Allowance mailed Sep. 26, 2012', 7 pgs. |  
67 | Non-Final Office Action Mailed Sep. 19, 2012 in Co-Pending U.S. Appl. No. 12/844,818 of Glover, R., filed Jul. 27, 2010. cited by applicant. |  
68 | U.S. Appl. No. 12/737,887, Notice of Allowance mailed Feb. 24, 2014, 9 pages. |  
69 | Office Action from U.S. Appl. No. 10/571,148 mailed Sep. 10, 2010. |  
70 | Japanese Patent Office, Office Action in App. No. 2011-524128, Aug. 25, 2013. |  
71 | Notice of Ailowanct dated Dec. 4, 2014; for U.S. Appl. No, 13/405,406; 11 pages. |  
72 | Communication from the Japanese Patent Office mailed Jul. 10, 2012, in Japanese application No. 2009-203514. |  
73 | Office Action issued Oct. 18, 2013, in Japanese Patent Application No. 2008-206291 (with English-language translation). |  
74 | Japanese Notice of Reasons for Rejection for Japanese Application No. 2010-032671; Date of Mailing: Aug. 20, 2013; with English Translation. |  
75 | U.S. Appl. No. 10/392,697, dated Jan. 5, 2007 Final Office Action. |  
76 | USPTO Jun. 27, 2013 Final Office Action from U.S. Appl. No. 12/968,163. |  
77 | U.S. Appl. No. 11/416,473, Response to Rule 312 Communication mailed Jun. 11, 2010. |  
78 | Office Action for U.S. Appl. No. 12/101,099 dated Apr. 19, 2010. |  
79 | Advisory Action of U.S. Appl. No. 10/613,608 mailed Aug. 4, 2005. |  
80 | Notice of Allowance dated Jan. 20, 2012, U.S. Appl. No. 12/349,513, filed Jan. 6, 2009. |  
81 | Notice of Rejection issued on Jul. 27, 2010 in related matter Japanese Patent Publication JP 2007-540191. English translation provided by Kashiwabara International Patent Bureau. |  
82 | U.S. Appl. No. 12/022,098 Office Action dated Mar. 16, 2009. |  
83 | Korean Notice of Allowance for KR Appl. No. 10-2011-7002566 dated Sep. 26, 2011. |  
84 | Non-Final Office Action for U.S. Appl. No. 09/753,982; Mailed on Nov. 9, 2004; 8 pages. |  
85 | Official Action for U.S. Appl. No. 11/782,569, mailed Oct. 12, 2010. |  
86 | English Translation of Japanese Patent Application No. 2013-083152 Office Action dated Oct. 21, 2014. |  
87 | Nov. 22, 2011 Japanese official action in connection with a counterpart Japanese patent application. |  
88 | Office Action for Swiss National Phase Patent Application No. 01183/14, mailed Mar. 24, 2016, 3 pages. |  
89 | Notice of Reasons for Rejection dated Aug. 9, 2016 corresponding to Japanese Patent Application No. 2013-012447 and English translation thereof. |  
90 | European Patent Application No. EP-06804122.7, Official Letter mailed Jan. 25, 2011. |  
91 | Notification of First Office Action in counterpart Chinese Patent Application No. 200880020777.7 dated Dec. 24, 2010. |  
92 | China Office Action for Application No. 200880000670.6, Aug. 5, 2010, Panasonic Corporation. |  
93 | 'U.S. Appl. No. 14/207,476, Non Final Office Action mailed Jun. 13, 2014'. |  
94 | Japanese Office Action dated Jul. 20, 2010 in Japanese Application No. 2004-167589. |  
95 | Official Communication for Australian Patent Application No. 2013237658 dated Feb. 2, 2015. |  
96 | “U.S. Appl. No. 15/072,484, Response filed Jul. 6, 2017 to Non Final Office Action dated Jan. 6, 2017”, 7 pgs. |  
97 | USPTO; Office Action forU.S. Appl. No. 12/154,465, mailed May 4, 2012. |  
98 | Final Office Action for U.S. Appl. No. 11/185,247, mailed Feb. 26, 2010. |  
99 | Second Office Action Issued in Chinese Patent Application No. 201110031217.2, Mailed Date: May 25, 2015, 6 Pages. |  
100 | Office Action from U.S. Appl. No. 11/888,251, dated Mar. 21, 2008.

</details>

### Open questions

- Need for further sub-classification? If yes, which sub-classes?
- Fields to be parsed?
- Corpus/ reference database for consolidation?

I leave these questions open to discussion.

Cheers

**Comment by gderasse** [comment]
Did you have a look at patents-public-data:uspto_oce_office_actions on BigQuery? It could provide useful food for thought.

**Comment by cverluise** [comment]
_Follow up_: The dataset is described in [USPTO Patent Prosecution Research Data: Unlocking Office Action Traits](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=3024621). 

> The Office Action Research Dataset for Patents contains detailed information derived from the Office actions issued by patent examiners to applicants during the patent examination process. The “Office action” is a written notification to the applicant of the examiner’s decision on patentability and generally discloses the grounds for a rejection, the claims affected, and the pertinent prior art. This initial release consists of three files derived from 4.4 million Office actions mailed during the **2008 to mid-2017 period from USPTO examiners** to the applicants of 2.2 million unique patent applications.

--> partial coverage, still can help shape the parsing strategy and address part of the consolidation task.

**Comment by cverluise** [comment]
Proposed parsing attributes (with names from aforementioned database)

- **application_id**: the unique application number + consolidation from grobid?
- **document_cd**: document code identifying the type of document issued by the examiner. For the USPTO, it can be `CTNF` (non final rejection) or `CTFR` (final rejection) -> we will certainly extend the possible categories
- **maild_dt**: mail date
